backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java

```diff
@@ -209,108 +191,115 @@ private ExperimentRun getExperimentRunFromRequest(
-                  artifactHandler.logArtifacts(
-                      newExperimentRun.getId(), newExperimentRun.getArtifactsList(), false));
-              futureLogs.add(
-                  featureHandler.logFeatures(
-                      newExperimentRun.getId(), newExperimentRun.getFeaturesList()));
-              futureLogs.add(
-                  codeVersionHandler.logCodeVersion(
-                      LogExperimentRunCodeVersion.newBuilder()
-                          .setId(newExperimentRun.getId())
-                          .setCodeVersion(newExperimentRun.getCodeVersionSnapshot())
-                          .setOverwrite(false)
-                          .build()));
+              if (!newExperimentRun.getTagsList().isEmpty()) {
+                futureLogs.add(
+                    tagsHandler.addTags(newExperimentRun.getId(), newExperimentRun.getTagsList()));
+              }
+              if (!newExperimentRun.getAttributesList().isEmpty()) {
+                futureLogs.add(
+                    attributeHandler.logKeyValues(
+                        newExperimentRun.getId(), newExperimentRun.getAttributesList()));
+              }
+              if (!newExperimentRun.getHyperparametersList().isEmpty()) {
+                futureLogs.add(
+                    hyperparametersHandler.logKeyValues(
+                        newExperimentRun.getId(), newExperimentRun.getHyperparametersList()));
+              }
+              if (!newExperimentRun.getMetricsList().isEmpty()) {
+                futureLogs.add(
+                    metricsHandler.logKeyValues(
+                        newExperimentRun.getId(), newExperimentRun.getMetricsList()));
+              }
+              if (!newExperimentRun.getObservationsList().isEmpty()) {
+                futureLogs.add(
+                    observationHandler.logObservations(
+                        newExperimentRun.getId(), newExperimentRun.getObservationsList(), now));
+              }
+              if (!newExperimentRun.getArtifactsList().isEmpty()) {
+                futureLogs.add(
+                    artifactHandler.logArtifacts(
+                        newExperimentRun.getId(), newExperimentRun.getArtifactsList(), false));
+              }
+              if (!newExperimentRun.getFeaturesList().isEmpty()) {
+                futureLogs.add(
+                    featureHandler.logFeatures(
+                        newExperimentRun.getId(), newExperimentRun.getFeaturesList()));
+              }
+              if (newExperimentRun.hasCodeVersionSnapshot()) {
+                futureLogs.add(
+                    codeVersionHandler.logCodeVersion(
+                        LogExperimentRunCodeVersion.newBuilder()
+                            .setId(newExperimentRun.getId())
+                            .setCodeVersion(newExperimentRun.getCodeVersionSnapshot())
+                            .setOverwrite(false)
+                            .build()));
+              }
```

Comment:
Added if condition if fields are empty then no need to create Future for it otherwise it will leading exceptions or bug inside handlers due to no data.